### PR TITLE
Make dependancy adal optional

### PR DESCRIPTION
**Pull request by iamneha**
Fixes issue :[kubernetes-client/python/issues#645]( https://github.com/kubernetes-client/python/issues/645)
To fix this issue:
1. remove `adal` from requirements
2. add adal to `extras_require` in setup.py
3. handle import error

Fix for 1 and 2 is [kubernetes-client/python/pull#668](https://github.com/kubernetes-client/python/pull/668)

### CODE REVIEW

File: config/kube_config.py [hunk 1]
```diff
@@ -218,21 +222,24 @@ def _load_azure_token(self, provider):
...
-            refresh_token, client_id, '00000002-0000-0000-c000-000000000000')
-
-        provider = self._user['auth-provider']['config']
-        provider.value['access-token'] = token_response['accessToken']
-        provider.value['expires-on'] = token_response['expiresOn']
-        if self._config_persister:
-            self._config_persister(self._config.value)
+        if 'adal' not in globals():
+            raise ImportError('refresh token error, adal library not imported')
+        else:
```
Comment (tomplus) [inline config/kube_config.py]: This `else` is not necessary here.

**Comment by tomplus** [comment]
BTW, there is a similar PR: https://github.com/kubernetes-client/python-base/pull/101

**Comment by micw523** [comment]
Per our slack discussion on Oct 31st, I asked you if you wanted to hand over the issue and you said yes. That was why I started to work on it and opened a PR.

If you want to work on it again, go straight ahead. I don’t mind - to me reviewing is just important as submitting a fix. However in the future please remember what you have said so that this situation does not happen any more.

**Comment by iamneha** [comment]
> Per our slack discussion on Oct 31st, I asked you if you wanted to hand over the issue and you said yes. That was why I started to work on it and opened a PR.

Yes @micw523 you asked me the issue you were working on but I think you are confusing this issue with another one you asked for 
This is the issue you were working on " [Loading yaml file](https://github.com/kubernetes-client/python/pull/655) "


> If you want to work on it again, go straight ahead. I don’t mind - to me reviewing is just important as submitting a fix. However in the future please remember what you have said so that this situation does not happen any more.

Please never start working on any issue without assigning yourself on that issue. I would have stopped working on it if you mentioned that you are working on it.

If you still want to work on it please assign yourself to that issue and I'm happy to help you with the issue :)

**Comment by dims** [comment]
@iamneha @micw523 i left a note in https://github.com/kubernetes-client/python-base/pull/101

**Pull request merged**